Fitness/Views/Headers/HomeSectionHeaderView.swift

```diff
@@ -21,25 +32,53 @@ class HomeSectionHeaderView: UICollectionReusableView {
         titleLabel = UILabel()
         titleLabel.font = ._14MontserratBold
         titleLabel.textColor = .fitnessDarkGrey
+        titleLabel.text = ""
         addSubview(titleLabel)
 
+        navigationButton = UIButton()
+        navigationButton.setTitleColor(.fitnessDarkGrey, for: .normal)
+        navigationButton.contentHorizontalAlignment = .right
+        navigationButton.titleLabel?.font = ._14LatoBlack
+        navigationButton.addTarget(self, action: #selector(viewAll), for: .touchUpInside)
+        addSubview(navigationButton)
+
         // MARK: - CONSTRAINTS
         titleLabel.snp.updateConstraints { make in
             make.leading.equalTo(16)
-            make.top.equalToSuperview()
-            make.height.equalTo(titleLabel.intrinsicContentSize.height)
+            make.top.trailing.equalToSuperview()
+            make.height.equalTo(20)
         }
-    }
-
-    func setTitle(title: String) {
-        titleLabel.text = title
 
-        titleLabel.snp.updateConstraints { make in
-            make.height.equalTo(titleLabel.intrinsicContentSize.height)
+        navigationButton.snp.updateConstraints { make in
+            make.trailing.equalToSuperview().offset(-16)
+            make.width.equalTo(60)
```

Comment:
make numbers a constant and move this to `setupConstraints` function just to be consistent